crates/nu-parser/tests/test_parser.rs

```diff
@@ -117,6 +117,99 @@ pub fn parse_binary_with_incomplete_binary_format() {
     assert_eq!(expressions[0].expr, Expr::Binary(vec![0b00000010]))
 }
 
+#[test]
+pub fn parse_binary_with_octal_format() {
+    let engine_state = EngineState::new();
+    let mut working_set = StateWorkingSet::new(&engine_state);
+
+    let (block, err) = parse(&mut working_set, None, b"0o[250]", true, &[]);
+
+    assert!(err.is_none());
+    assert!(block.len() == 1);
+    let expressions = &block[0];
+    assert!(expressions.len() == 1);
+    assert_eq!(expressions[0].expr, Expr::Binary(vec![0o250]))
+}
+
+#[test]
+pub fn parse_binary_with_incomplete_octal_format() {
+    let engine_state = EngineState::new();
+    let mut working_set = StateWorkingSet::new(&engine_state);
+
+    let (block, err) = parse(&mut working_set, None, b"0o[2]", true, &[]);
+
+    assert!(err.is_none());
+    assert!(block.len() == 1);
+    let expressions = &block[0];
+    assert!(expressions.len() == 1);
+    assert_eq!(expressions[0].expr, Expr::Binary(vec![0o2]))
+}
+
+#[test]
+pub fn parse_binary_with_invalid_octal_format() {
+    let engine_state = EngineState::new();
+    let mut working_set = StateWorkingSet::new(&engine_state);
+
+    let (block, err) = parse(&mut working_set, None, b"0b[90]", true, &[]);
+
+    assert!(err.is_none());
+    assert!(block.len() == 1);
+    let expressions = &block[0];
+    assert!(expressions.len() == 1);
+    assert_eq!(
+        expressions[0].expr,
+        Expr::ExternalCall(
+            Box::new(Expression {
+                expr: Expr::String("0b[90]".to_string()),
+                span: Span { start: 0, end: 6 },
+                ty: Type::String,
+                custom_completion: None
+            }),
+            vec!()
+        )
+    )
```

Comment:
Don't know if we want to commit to this behavior with a test (discussion in #5155 and originally #5149). 
From a pleasant user experience perspective a malformed literal may better return an informative error than fall back to `ExternalCall` which can be signified explicitly with `^name_of_external`. (tricky if something that looks like a malformed nu literal is expected to work as an argument to an external as an unquoted string).
Maybe just test that `assert!(!matches(expressions[0].expr, Expr::Binary(val)))`

Reply:
fixed

Reply:
Thx